pageserver/src/tenant/gc_block.rs

```diff
@@ -1,10 +1,16 @@
-use std::collections::HashMap;
+use std::{collections::HashMap, time::Duration};
 
+use super::{remote_timeline_client::index::GcBlockingReason, tasks};
+use tokio_util::sync::CancellationToken;
 use utils::id::TimelineId;
 
-use super::remote_timeline_client::index::GcBlockingReason;
+type TimelinesBlocked = HashMap<TimelineId, enumset::EnumSet<GcBlockingReason>>;
 
-type Storage = HashMap<TimelineId, enumset::EnumSet<GcBlockingReason>>;
+#[derive(Default)]
+struct Storage {
+    pub timelines_blocked: TimelinesBlocked,
+    pub tenant_blocked: bool,
```

Comment:
I wonder if we should instead of tenant_blocked just put the concrete reason, but of course, I cannot name "the time we need to spend blocked after transitioning to AttachedSingle"